objectherkenning_openbare_ruimte/frame_extraction/frame_extraction.py

```diff
@@ -18,11 +21,96 @@
 settings = ObjectherkenningOpenbareRuimteSettings.get_settings()
 
 
+class FFMPEG_frame_extractor:
+    """
+    This class wraps around the ffmpeg command to facilitate frame extraction and
+    distortion correction from videos.
+
+    Checks are in place to ensure arguments are valid:
+    fps, cx, cy, k1, k2 must be float
+    input_file must be an existing file
+    output_folder must be an existing folder
+    log_file must point to a valid filename in an existing folder
+    """
+
+    def __init__(self, fps, distortion_settings):
+        self.cmd = None
+
+        cx = distortion_settings["cx"]
+        cy = distortion_settings["cy"]
+        k1 = distortion_settings["k1"]
+        k2 = distortion_settings["k2"]
+
+        if not isinstance(fps, float):
+            print(f"Invalid type for fps: {type(fps)}")
+            return
+        if not isinstance(cx, float):
+            print(f"Invalid type for fps: {type(fps)}")
+            return
+        if not isinstance(cy, float):
+            print(f"Invalid type for fps: {type(fps)}")
+            return
+        if not isinstance(k1, float):
+            print(f"Invalid type for fps: {type(fps)}")
+            return
+        if not isinstance(k2, float):
+            print(f"Invalid type for fps: {type(fps)}")
+            return
+
+        # Prepare FFMPEG command for distortion correction and output quality
+        self.filter_str = (
+            f"fps={fps},lenscorrection=cx={cx}:cy={cy}:k1={k1}:k2={k2}:i=bilinear"
+        )
+
+    def create_command(self, input_file: str, output_path: str, log_file: str) -> str:
+        if not os.path.isfile(input_file):
+            print(f"Input file not found: {input_file}")
+            return None
+        if not os.path.isdir(os.path.dirname(output_path)):
+            print(f"Output folder not found: {os.path.dirname(output_path)}")
+            return None
+        if not is_valid_filepath(output_path):
+            print(f"Output file path not valid: {output_path}")
+            return None
+        if not os.path.isdir(os.path.dirname(log_file)):
+            print(f"Log folder not found: {os.path.dirname(log_file)}")
+            return None
+        if not is_valid_filepath(log_file):
+            print(f"Log file path not valid: {log_file}")
+            return None
```

Comment:
Same here, instead of printing we could raise exceptions in a try-except block:

`try:`
`   if not os.path.isfile(input_file):`
`      raise FileNotFoundError(f"Input file not found: {input_file}")`
`...`
`   if not is_valid_filepath(log_file):`
`      raise ValidationError(f"Log file path not valid: {log_file}")`
`except (FileNotFoundError, ValidationError) as e:`
`   log.error(e)`